src/generators/android/android.ts

```diff
@@ -0,0 +1,328 @@
+	return `public Builder ${name}(${modifiers} ${type} ${name})`
+}
+
+function generateBuilderFunctionBody(name: string, rawName: string, type: string): string {
+	const isArrayType = type.match(/List\<(.*)\>/)
+	const isSerializable = Object.values(JavaType).every(t => t !== type)
+
+	const defaultHandler = (raw = rawName, n = name, indentFirstLine = true) =>
+		`${indentFirstLine ? Separator.Indent : ''}properties.putValue("${raw}", ${n});\n` +
+		`${Separator.NewLineIndent}return this;`
+
+	const serializeObject =
+		`${Separator.Indent}if(${name} instanceof SerializableProperties){\n` +
+		`${Separator.NewLineIndent}${
+			Separator.Indent
+		}properties.putValue("${rawName}", ((SerializableProperties) ${name}).toProperties());\n` +
+		`${Separator.NewLineIndent}}else{\n` +
+		`${Separator.NewLineIndent}${Separator.Indent}properties.putValue("${rawName}", ${name});\n` +
+		`${Separator.NewLineIndent}}\n` +
+		`${Separator.NewLineIndent}return this;`
+
+	const serializeArray =
+		`${Separator.Indent}List<?> p = TypewriterUtils.serialize(${name});\n` +
+		`${Separator.NewLineIndent}${defaultHandler(rawName, 'p', false)}`
+
+	return isArrayType && isArrayType[1] !== 'Properties'
+		? serializeArray
+		: isSerializable
+		? serializeObject
+		: defaultHandler()
+}
+
+const getValidParams = (potentialParams: Param[]) =>
+	potentialParams.reduce((acc: string[], { hasParam, name, type }) => {
+		if (hasParam) {
+			acc.push(`${Modifier.FinalNullable} ${type} ${name}`)
+		}
+		return acc
+	}, [])
+
+const getValidArgs = (potentialArgs: Arg[], defaultArg?: string[]) =>
+	potentialArgs.reduce((acc: string[], { inUse, execution, fallback }) => {
+		if (inUse) {
+			acc.push(execution)
+		} else if (fallback) {
+			acc.push(fallback)
+		}
+		return acc
+	}, defaultArg || [])
+
+const intersperse = (values: string[], char: string) => {
+	let joined = values[0] || ''
+	if (values.length) {
+		for (let i = 1; i < values.length; i++) {
+			joined += `${char}${values[i]}`
+		}
+	}
+	return joined
+}
+
+function generateFunctionSignature(
+	{ functionName, propsParam }: { functionName: string; propsParam: boolean },
+	withOptions: boolean
+): string {
+	const params = getValidParams([
+		{ hasParam: propsParam, name: 'props', type: upperFirst(functionName) },
+		{ hasParam: withOptions, name: 'options', type: 'Options' },
+	])
+
+	return `public void ${functionName}(${intersperse(params, Separator.Comma)})`
+}
+
+function generateFunctionExecution(
+	{ rawEventName, propsParam }: { rawEventName: string; propsParam: boolean },
+	withOptions: boolean
+): string {
+	const args = getValidArgs(
+		[
+			{ inUse: propsParam, execution: Properties.ToProperties, fallback: Properties.Create },
+			{ inUse: withOptions, execution: Options.MergeOptions, fallback: Options.Create },
+		],
+		[`"${rawEventName}"`]
+	)
+	return `{
+    this.analytics.track(${intersperse(args, Separator.Comma)});
```

Comment:
Think you can remove `intersperse` in favor of `Array.prototype.join`:

```suggestion
    this.analytics.track(${args.join(Separator.Comma)});
```

Reply:
ha yep - I remember doing this for a reason which now completely eludes me.

Reply:
haha happens to me too. I figured you were originally using `intersperse` as a handlebars function :)